doc/development/guide/logging.rst

```diff
@@ -0,0 +1,419 @@
+directions (see
+`here <https://docs.python.org/3/library/logging.handlers.html>`__ for
+more info), but for this example we have defined stream handlers
+(printing to the screen via the standard output), and a file handler
+with a size cap at 16MB. Each handler can be customized by filters and
+formatters so that the consumed message fits the needs of the user.
+
+.. code:: toml
+
+   ###############################################################################
+   # Define logger controls for internal and external packages
+   ###############################################################################
+
+   [loggers]
+
+   # Control JAX logging 
+   [loggers.jax]
+   handlers = ["qml_debug_stream",]
+   level = "DEBUG"
+   propagate = false
+
+   # Control logging in the executing Python script
+   [loggers.__main__]
+   handlers = ["qml_debug_stream",]
+   level = "DEBUG"
+   propagate = false
+
+   # Control logging across pennylane
+   [loggers.pennylane]
+   handlers = ["qml_debug_stream",]
+   level = "DEBUG" # Set to 1 for highest verbosity
+   propagate = false
+
+   # Control logging specifically in the pennylane.qnode module
+   # Note the required parenthesis to overcome TOML nesting issues
+   [loggers."pennylane.qnode"]
+   handlers = ["qml_debug_stream_alt",]
+   level = "DEBUG" # Set to 1 for highest verbosity
+   propagate = false
+
+   ###############################################################################
+
+Finally, the ``loggers`` section which controls the individual loggers
+across the packages we are using. Python’s logging framework follows a
+parent-child hierarchy, where a logging configuration set at a parent
+level will set all child levels with the same features. In this
+instance, we have configured JAX, PennyLane and our script to all log
+into the ``qml_debug_stream`` handler we defined earlier, and modified
+the child logger ``"pennylane.qnode"`` (parenthesis needed due to TOML
+parsing limitations) to use a different logger, in this case
+``qml_debug_stream_alt``. We are free to define the module/package
+log-level here (we opt for ``DEBUG`` for all), and to also use multiple
+handlers per logger (such as for logging to the standard output and
+files through ``qml_debug_stream`` and ``qml_debug_file``
+simultaneously). Given the complexity explosion with configuring these
+options, the default features in ``log_config.toml`` all use the same
+log-level, and handler, which can be adjusted based on developer needs.
+
+Logging example with PennyLane and JAX’s JIT support
+----------------------------------------------------
+
+As mentioned above, we have added execution function entry logging
+supports, including some supports for each target interface. We can
+examine this support for both internal and external packages, where we
+enable logs for JAX, which has support for Python-native log messages.
+To enable logging specifcally for JAX, we can modify the ``level``
```

Comment:
```suggestion
To enable logging specifically for JAX, we can modify the ``level``
```
also, you say "we can modify the `level`", but use the same level value as above. is that intentional?

Reply:
In this case yes, it is intentional. JAX's logger level is not set to the same one as the PennyLane loggers. This is showing we can also set JAX's log level to match that of PennyLane. 

I'll think about how to express this in a way to better declare intent.

Reply:
to be clear, I mean that [the 4 lines below](https://github.com/PennyLaneAI/pennylane/blob/f4269b8ad25a1ac26f9664713cdf071c42394bbc/doc/development/guide/logging.rst?plain=1#L256-L259) are the same as [the 4 lines above](https://github.com/PennyLaneAI/pennylane/blob/f4269b8ad25a1ac26f9664713cdf071c42394bbc/doc/development/guide/logging.rst?plain=1#L201-L204). Are you talking about Jax's own logger (jax, and not pennylane/interfaces/jax)? I thought these configs change the logger for PennyLane's interface to jax, and not for jax itself. If it's the latter, how do we know the logger name is just `loggers.jax`? I'm now guessing it is for jax itself, since our interface's `__name__` would be "pennylane.interfaces.jax"...

Reply:
This is controlling the behaviour of the jax internal logger, not our jax interface.
We can control any external package log-level and behaviour in this way.

All logging is hierarchical in Python, so assuming there exists some nested logger in the package, setting the log level at the package name will override all internals, unless explicitly expressed otherwise. In this way, just using the `jax` package name for the logger is enough to enable debug level log messages to propagate up.

Reply:
Ah, apologies, I see what you mean now. That earlier JAX entry should be WARN. I'll update.